docs/user/unsupported/mbstyle.rst

```diff
@@ -19,14 +24,544 @@ Quick example of module usage (API change):
+
+    .. code-block:: json
+
+      [
+        "all",
+        ["==", "class", "street_limited"],
+        [">=", "admin_level", 3],
+        ["!in", "$type", "Polygon"]
+      ]
+
+MapBox Style Grammar
+^^^^^^^^^^^^^^^^^^^^
+  *JSON does not allow for comments within the data therefore comments will be noted through the placement of the comment between open < and close > angle brackets. All properties are optional unless otherwise noted as Requried*
+
+  Root Properties
+  '''''''''''''''
+
+  .. code-block:: json
+
+    {
+      "version": 8, <Required>
+      "name": "Mapbox Streets",
+      "sprite": "mapbox://sprites/mapbox/streets-v8",
+      "glyphs": "mapbox://fonts/mapbox/{fontstack}/{range}.pbf",
+      "sources": {...}, <Required>
+      "layers": [...] <Required>
+    }
+
+  layers
+  ''''''
+  Layers are drawn in the order they appear in the layer array. Layers have two additional properties that determine how data is rendered: *layout* and *paint*
+
+  Background Layer definition
+  '''''''''''''''''''''''''''
+  .. code-block:: json
+
+    {
+      "layers" : [
+        {
+          "id": "backgroundcolor",
+          "type": "background",
+          "source": "test-source",
+          "source-layer": "test-source-layer",
+          "layout": {
+            "visibility": "visible"
+          },
+          "paint": {
+            "background-opacity": 0.45,
+            "background-color": "#00FF00"
+          }
+        }
+      ]
+    }
+
+  *background-color* is disabled by the presence of *background-pattern*
+
+  Fill Layer Definition
+  '''''''''''''''''''''
+  .. code-block:: json
+
+    {
+      "layers": [
+        {
+          "id": "testid",
+          "type": "fill",
+          "source": "geoserver-states",
+          "source-layer": "states",
+          "layout": {
+            "visibility": "visible"
+          },
+          "paint": {
+            "fill-antialias": "true",
+            "fill-opacity": 0.84,
+            "fill-color": "#FF595E",
+            "fill-outline-color":"#1982C4",
+            "fill-translate": [20,20],
+            "fill-translate-anchor": "map",
+            "fill-pattern": <String>
+          }
+        }
+      ]
+    }
+
+  Line Layer Definition
+  '''''''''''''''''''''
+
+  .. code-block:: json
+
+    {
+      "layers": [
+        {
+          "id": "test-id",
+          "type": "line",
+          "source": "test-source",
+          "source-layer": "test-source-layer",
+          "layout": {
+              "line-cap": "square",
+              "line-join": "round",
+              "line-mitre-limit": 2, <Optional - Requires line-join=mitre>
```

Comment:
we have a different kind of comment here, please remove or confirm if json supports comments.